src/Mutex/Impl/Internal/MutexWorkflow.workflow.cs

```diff
@@ -0,0 +1,55 @@
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.Mutex.Impl.Internal;
+
+[Workflow]
+internal class MutexWorkflow
+{
+    private readonly ILockHandler lockHandler = WorkflowMutex.CreateLockHandler();
```

Comment:
Not sure there is value for a separate lock handler concept instead of just embedding logic in this class which is the only place that uses it

Reply:
Here a separate object is used, which is returned from WorkflowMutex for the reason that signals need to be sent and they are not strongly typed. Therefore, I thought that it would be better for all internal communication to be centralized in WorkflowMutex. In general, when I was writing this sample, I constantly felt a contradiction, that in some places we have strongly typed calls, while others do not have such capability

Reply:
Hrmm, yeah it makes sense to me since you're sharing those names. I think I was just hoping to have to jump around to fewer files for this simple sample to see what things do.